client/app/src/main/java/com/example/criengine/Activities/BookActivity.java

```diff
@@ -38,5 +42,10 @@ protected void onCreate(Bundle savedInstanceState) {
         bookTitle.setTag(bookTitle.getKeyListener());
         bookDetail.setTag(bookDetail.getKeyListener());
         bookAuthor.setTag(bookAuthor.getKeyListener());
+        customOnCreate();
     }
+
+    abstract protected void customOnCreate();
+
+    abstract protected void inflate();
```

Comment:
could you explain what this method does?

Reply:
So by inheriting from an abstract class it will cause the nav-bar fragment to throw nullpointer errors. Something along the lines of the fragment not being inflated by the time it was being called (you were setting the layout before calling the super). So by doing it in this format, you still call the super class and get all its attributes, but now we can set the inflator setup in the start.

Reply:
@rjmah1 will need to do this with profile activity too since i saw he was doing stuff with abstract classes

Reply:
ah okay. thanks